includes/classes/traits/EventManager.php

```diff
@@ -13,17 +13,41 @@ trait EventManager
     public function notify($eventID, $param1 = array(), & $param2 = null, & $param3 = null, & $param4 = null, & $param5 = null, & $param6 = null, & $param7 = null, & $param8 = null, & $param9 = null)
     {
         $this->logNotifier($eventID, $param1, $param2, $param3, $param4, $param5, $param6, $param7, $param8, $param9);
-        $observers = & self::getStaticObserver();
+
+
+        $observers = &self::getStaticObserver();
         if (is_null($observers)) {
             return;
         }
-        foreach($observers as $key=>$obs) {
-            if ($obs['eventID'] == $eventID || $obs['eventID'] === '*') {
-                $method = 'update';
-                $testMethod = $method . self::camelize(strtolower($eventID), TRUE);
-                if (method_exists($obs['obs'], $testMethod))
-                    $method = $testMethod;
-                $obs['obs']->{$method}($this, $eventID, $param1,$param2,$param3,$param4,$param5,$param6,$param7,$param8,$param9);
+        foreach ($observers as $key => $obs) {
+            // identify the event
+            $actualEventId = $eventID;
+            $matchMap = [$eventID, '*'];
```

Comment:
Shouldn't these two be in advance/outside of the `foreach` loop?

Reply:
If it were permanently standing alone, then yes.

But this is a near-clone of code in the `base` class where alias-matching happens right here as well, and in the execution of this outer loop those variables may (I think) get updated by the inner alias-detection loop, so this reset has benefit.